libc/string/string_tokpbrkstr.c

```diff
@@ -0,0 +1,875 @@
+
+	/* 256 is not taken as a token because on element after it is a null term and it is recognized as the end of set */
+	TEST_ASSERT_EQUAL_PTR(NULL, strtok_r(NULL, divider, &rest));
+	TEST_ASSERT_NOT_NULL(rest);
+	TEST_ASSERT_EMPTY(rest);
+
+	free(extAsciiStr);
+	extAsciiStr = create_extAscii_set();
+	/*
+	 * Checking available tokenization on all elements
+	 * from outside of ASCII charset one by one, with
+	 * the first separator set as the first element of the set
+	 */
+	TEST_ASSERT_EQUAL_PTR(extAsciiStr, strtok_r(extAsciiStr, "\128", &rest));
+	for (i = ASCII_LENGTH + 1; i < sizeof(separator); i++) {
+		separator[i - ASCII_LENGTH] = i;
+		TEST_ASSERT_EQUAL_PTR(&extAsciiStr[i - ASCII_LENGTH], strtok_r(NULL, &separator[i - ASCII_LENGTH], &rest));
+		TEST_ASSERT_NOT_NULL(rest);
+	}
+
+	free(extAsciiStr);
+#endif
+}
+
+
+TEST(string_tok_r, state_restriction)
+{
+	/* Ifdef used because lack of function 'strtok_r' */
+#ifdef __phoenix__
+	TEST_IGNORE();
+#else
+
+	char *rest[ASCII_LENGTH] = { 0 },
+		 *tok,
+		 *asciiStr = testdata_createCharStr(ASCII_LENGTH);
+	int i;
+
+
+	for (i = 1; i < ASCII_LENGTH - 1; i++) {
+		TEST_ASSERT_EQUAL_PTR(&asciiStr[1], tok = strtok_r(&asciiStr[1], &asciiStr[i + 1], &rest[i]));
+		TEST_ASSERT_EQUAL_STRING(&asciiStr[1], tok);
+		TEST_ASSERT_NOT_NULL(rest[i]);
+		/*We must avoid the first element to fill up the element behind for assertions*/
+		if (i > 2) {
+			TEST_ASSERT_NOT_NULL(rest[i - 1]);
+			TEST_ASSERT_EQUAL_HEX64(rest[i - 1], rest[i]);
+		}
+	}
+
+	free(asciiStr);
+#endif
+}
+
+
+/*
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_str)
+{
+}
+
+
+TEST_TEAR_DOWN(string_str)
+{
+}
+
+
+TEST(string_str, basic)
+{
+	const char loremIpsum[BUFFSIZE] = LOREM_IPSUM;
+	char *asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	/* Standard use of strstr on arrays */
+	TEST_ASSERT_EQUAL_PTR(loremIpsum, strstr(loremIpsum, "Lorem"));
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[6], strstr(loremIpsum, "Ipsum"));
+	TEST_ASSERT_EQUAL_PTR(&loremIpsum[12], strstr(loremIpsum, "Dolor"));
+
+	/* Simple usage of strstr with ascii set */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strstr(asciiStr, asciiStr));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 4], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 4]));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 3], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 3]));
+	TEST_ASSERT_EQUAL_PTR(&asciiStr[ASCII_LENGTH / 2], strstr(asciiStr, &asciiStr[ASCII_LENGTH / 2]));
+
+
+	free(asciiStr);
+}
+
+
+TEST(string_str, empty_args)
+{
+	const char empty[] = "",
+			   str[] = "Lorem";
+
+
+	char *asciiStr = testdata_createCharStr(ASCII_LENGTH);
+
+	/* Different scenarios of using empty input or output*/
+	TEST_ASSERT_EQUAL_PTR(NULL, strstr(empty, str));
+	/* Passing an empty string as the second parameter to get first */
+	TEST_ASSERT_EQUAL_PTR(asciiStr, strstr(asciiStr, ""));
+
+	TEST_ASSERT_EQUAL_PTR(str, strstr(str, str));
```

Comment:
How is it related with empty args?

Reply:
Moreover I see that the similar case is already added in `basic`, so it's not needed to leave it.